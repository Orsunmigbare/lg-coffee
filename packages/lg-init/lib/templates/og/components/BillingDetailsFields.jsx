import React from 'react';
import Form from 'react-bootstrap/Form';

const BillingDetailsFields = () => (
  <div className="d-flex flex-column align-items-center">
    <div style={{ maxWidth: '80%' }}>
      <Form.Row>
        <h2 className="mt-5">Billing Information</h2>
        <Form.Control
          className="mt-3"
          type="text"
          label="name"
          name="name"
          placeholder="Jane Doe"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="mt-3"
          type="text"
          label="email"
          name="email"
          placeholder="jane.doe@example.com"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="mt-3"
          type="text"
          label="address"
          name="address"
          placeholder="185 Berry St. Suite 550"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="mt-3"
          type="text"
          label="city"
          name="city"
          placeholder="San Francisco"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="mt-3"
          type="text"
          label="state"
          name="state"
          placeholder="CA"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="my-3"
          type="text"
          label="zip"
          name="zip"
          placeholder="94103"
          required
          style={{
            border: '0',
          }}
        />
      </Form.Row>
    </div>
  </div>
);

export default BillingDetailsFields;
