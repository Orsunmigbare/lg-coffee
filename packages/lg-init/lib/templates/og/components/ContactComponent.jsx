/* eslint-disable react/require-default-props */
import React, {useState} from 'react';
import Form from 'react-bootstrap/Form';
import ContactFrom from './ContactForm';



const Contact = () => {
  
  const [isProcessing, setProcessingTo] = useState(false);

  const handleFormSubmit = async (ev) => {
    ev.preventDefault();

    const messageDetails = {
      name: ev.target.name.value,
      email: ev.target.email.value,
      message: ev.target.message.value,
    };

    setProcessingTo(true);
    console.log(messageDetails)
  }
    // // create payment intent
    // const { data: clientSecret } = await axios.post('/api/payment_intents', {
    //   amount: price,
    // });

  return (  
    <Form onSubmit={handleFormSubmit}>
      <ContactFrom />
      <button
        type="submit"
        className="btn btn-outline-dark mb-5 mt-2 mx-4"
      >
        Send Message
      </button>
    </Form>
  )
}


export default Contact;
