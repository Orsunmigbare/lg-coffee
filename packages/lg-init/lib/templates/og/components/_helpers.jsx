/* eslint-disable react/require-default-props */
import React, { useState } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import Modal from 'react-bootstrap/Modal';

const defaultTheme = {
  text: { color: '#212529' },
  secondaryText: { color: '#6C757D' },
  primaryBackground: { backgroundColor: '#F8F9FA' },
  secondaryBackground: { backgroundColor: '#FFFFFF' },
};

const H1 = ({ children, className, theme }) => (
  <h1 className={clsx(className, theme)}>{children}</h1>
);

H1.propTypes = {
  children: PropTypes.node.isRequired,
  // applies to general styles
  className: PropTypes.string.isRequired,
  // applies to color
  theme: PropTypes.string.isRequired,
};

const H2 = ({ children, className, theme }) => (
  <h2 className={clsx(className, theme)}>{children}</h2>
);

H2.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

const H3 = ({ children, className, theme }) => (
  <h3 className={clsx(className, theme)}>{children}</h3>
);

H3.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

const H4 = ({ children, className, theme }) => (
  <h4 className={clsx(className, theme)}>{children}</h4>
);

H4.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

const H5 = ({ children, className, theme }) => (
  <h5 className={clsx(className, theme)}>{children}</h5>
);

H5.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

const H6 = ({ children, className, theme }) => (
  <h6 className={clsx(className, theme)}>{children}</h6>
);

H6.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
};

const Pill = ({ title, className, theme, handleShow }) => (
  <button
    type="button"
    onClick={handleShow}
    className={clsx('btn btn-outline-dark mr-2 mt-2 lg_pill', className)}
    style={theme}
  >
    {title}
  </button>
);

Pill.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  theme: PropTypes.string.isRequired,
  handleShow: PropTypes.func.isRequired,
};

const PrimaryModal = ({ children, className, theme, title }) => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Pill
        title={title}
        className={className}
        handleShow={handleShow}
        style={theme.secondaryBackground || defaultTheme.secondaryBackground}
      />
      <Modal show={show} onHide={handleClose}>
        <button
          type="button"
          onClick={handleClose}
          className="align-self-end mt-4 mr-4 btn btn-outline-dark"
          style={{
            border: 'none',
            background: 'none',
            fontSize: '16px',
            fontFamily: 'Montserrat',
          }}
        >
          Go Back
        </button>
        {children}
      </Modal>
    </>
  );
};

PrimaryModal.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  theme: PropTypes.string,
  title: PropTypes.string.isRequired,
};

const Wrap = ({ children, className }) => (
  <div className={clsx(className, 'd-flex flex-wrap')}>{children}</div>
);

Wrap.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
};

export { H1, H2, H3, H4, H5, H6, Pill, PrimaryModal, Wrap };
