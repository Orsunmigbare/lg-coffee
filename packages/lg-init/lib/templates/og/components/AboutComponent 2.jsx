/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import { H4, P } from './_helpers';

const AboutComponent = ({ store, theme }) => (
  <>
    <H4 theme={theme}>About Us</H4>
    <P theme={theme}>{store.about || 'Coming Soon'}</P>
  </>
);

AboutComponent.propTypes = {
  store: PropTypes.shape({
    about: PropTypes.string,
  }),
  theme: PropTypes.shape({}),
};

export default AboutComponent;
