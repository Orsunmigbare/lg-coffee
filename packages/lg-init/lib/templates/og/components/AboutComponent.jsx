/* eslint-disable react/require-default-props */
import React from 'react';
import PropTypes from 'prop-types';
import { H4, P } from './_helpers';

const AboutComponent = ({ store }) => (
  <>
    <H4>About Us</H4>
    <P>{store.about || 'Coming Soon'}</P>
  </>
);

AboutComponent.propTypes = {
  store: PropTypes.shape({
    about: PropTypes.string,
  }),
};

export default AboutComponent;
