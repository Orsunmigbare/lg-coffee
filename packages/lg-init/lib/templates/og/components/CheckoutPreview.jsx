import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Loading from './Loading';
import { formatUSD } from '../utils/formatUSD';
import { H2, H3, H4 } from './_helpers';

const CheckoutPreview = ({ cart }) => {
  if (!cart) <Loading data={cart} />;

  const [cartInStorage, setCartInStorage] = useState(cart);
  const [isCartEmpty, setCartToEmpty] = useState();

  useEffect(() => {
    // eslint-disable-next-line no-unused-expressions
    cartInStorage ? setCartToEmpty(false) : setCartToEmpty(true);
    
  }, [cartInStorage]);

  const handleRemoveFromCart = (id) => {
    const updatedCart = cartInStorage;

    delete updatedCart[id];

    setCartInStorage(updatedCart);

    if (Object.keys(updatedCart).length === 0) {
      localStorage.removeItem('cart');
      setCartToEmpty(true);
    } else localStorage.setItem('cart', JSON.stringify(cartInStorage));
  };

  return (
    <>
      <Container>
        {cartInStorage === undefined ||
        cartInStorage === null ||
        isCartEmpty ? (
          <H2 className="m-4">No Products</H2>
        ) : (
          <>
            <Row className="d-flex align-items-center">
              <H3 className="ellipse pb-5 pl-5">In Cart</H3>
            </Row>

            {Object.keys(cartInStorage).map((i) => (
              <Row
                key={cartInStorage[i].sku}
                className="d-flex align-items-center rounded px-5"
              >
                <Col xs={4}>
                  <img
                    src={cartInStorage[i].images[0]}
                    alt={cartInStorage[i].name}
                    style={{ height: '100px', width: '100px' }}
                  />
                </Col>
                <Col xs={8} className="d-flex flex-column">
                  <Row className="align-self-end mr-2">
                    <H4
                      style={{ width: '100%', margin: '0' }}
                      className="ellipse"
                    >
                      {cartInStorage[i].name}
                    </H4>
                  </Row>
                  <Row className="align-self-end align-items-end mr-2">
                    <p className="mt-2" style={{ margin: '0', padding: '0' }}>
                      ${formatUSD(cartInStorage[i].price)} USD
                    </p>
                    <p
                      className="ml-4 mr-2"
                      style={{ margin: '0', padding: '0' }}
                    >
                      Qty: {cartInStorage[i].qty}
                    </p>
                    <button
                      type="button"
                      style={{
                        border: 'none',
                        background: 'none',
                        margin: '0',
                        padding: '0',
                      }}
                      className="btn btn-outline-danger font-weight-bolder ml-4"
                      onClick={() => {
                        handleRemoveFromCart(i);
                      }}
                    >
                      X
                    </button>
                  </Row>
                </Col>
              </Row>
            ))}
          </>
        )}
      </Container>
    </>
  );
};

CheckoutPreview.propTypes = {
  // eslint-disable-next-line react/require-default-props
  cart: PropTypes.shape({
    images: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.string,
    qty: PropTypes.string,
  }),
};

export default CheckoutPreview;
