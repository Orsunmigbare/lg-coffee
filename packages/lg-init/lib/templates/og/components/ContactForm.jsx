import React from 'react';
import PropTypes from 'prop-types';
import Form from 'react-bootstrap/Form';
import { H2 } from './_helpers';

const ContactForm = ({ theme }) => (
  <div className="d-flex flex-column align-items-center">
    <div style={{ maxWidth: '80%' }}>
      <Form.Row>
        <H2 className="mt-4" theme={theme}>
          Get In Touch
        </H2>

        <Form.Control
          className="mt-3"
          type="text"
          label="name"
          name="name"
          placeholder="Jane Doe"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="mt-3"
          type="text"
          label="email"
          name="email"
          placeholder="jane.doe@example.com"
          required
          style={{
            border: '0',
          }}
        />

        <Form.Control
          className="my-3"
          type="text"
          as="textarea"
          label="body"
          name="body"
          placeholder="Inquiring about my recent purchase..."
          required
          style={{
            border: '0',
          }}
        />

        <Form.Check
          className="my-3"
          type="checkbox"
          label="I am not a robot"
          name="body"
          placeholder="Your message here"
          required
          style={{
            border: '0',
          }}
        />
      </Form.Row>
    </div>
  </div>
);

ContactForm.propTypes = {
  // eslint-disable-next-line react/require-default-props
  theme: PropTypes.shape({}),
};

export default ContactForm;
