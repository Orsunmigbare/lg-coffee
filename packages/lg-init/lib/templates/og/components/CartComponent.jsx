/* eslint-disable no-unused-expressions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
import Router from 'next/router';
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { calc } from 'lg-calc';
import Modal from 'react-bootstrap/Modal';
import CheckoutForm from './CheckoutForm';
import CheckoutPreview from './CheckoutPreview';

const CartComponent = ({ cart }) => {
  const [isCartEmpty, setCartToEmpty] = useState();
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  useEffect(() => {
    cart ? setCartToEmpty(false) : setCartToEmpty(true);
  }, [cart]);

  return (
    <>
      <div variant="primary" onClick={handleShow}>
        <img
          src="/cart.png"
          alt="Checkout"
          style={{
            height: '20px',
            cursor: 'pointer',
          }}
        />
      </div>

      <Modal show={show} onHide={handleClose}>
        {isCartEmpty ? (
          <h2 className="m-4">Nothing in cart</h2>
        ) : (
          <div className="d-flex flex-column align-items-center">
            <div>
              <div className="d-flex flex-column">
                <button
                  type="button"
                  onClick={handleClose}
                  className="align-self-end mt-4 mr-4 btn btn-outline-dark"
                  style={{
                    border: 'none',
                    background: 'none',
                    fontSize: '16px',
                    fontFamily: 'Montserrat',
                  }}
                >
                  Go Back
                </button>
              </div>
              <CheckoutPreview cart={cart} />
              <CheckoutForm
                cart={cart}
                price={calc(cart)}
                onSuccessfulCheckout={() =>
                  Router.push('/checkout/order-confirmation')
                }
              />
            </div>
          </div>
        )}
      </Modal>
    </>
  );
};

CartComponent.propTypes = {
  // eslint-disable-next-line react/require-default-props
  cart: PropTypes.shape({}),
  data: PropTypes.shape({}).isRequired,
};

export default CartComponent;
