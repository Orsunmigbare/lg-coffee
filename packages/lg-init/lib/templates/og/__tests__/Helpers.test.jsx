import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { H1, H2, H3, H4, H5, H6, P } from '../components/_helpers';


it('H1 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H1>Hello World</H1>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H1 component renders unchanged', () => {
  const tree = renderer.create(<H1>Hello World</H1>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('H2 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H2>Hello World</H2>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H2 component renders unchanged', () => {
  const tree = renderer.create(<H2>Hello World</H2>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('H3 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H3>Hello World</H3>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H3 component renders unchanged', () => {
  const tree = renderer.create(<H3>Hello World</H3>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('H4 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H4>Hello World</H4>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H4 component renders unchanged', () => {
  const tree = renderer.create(<H4>Hello World</H4>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('H5 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H5>Hello World</H5>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H5 component renders unchanged', () => {
  const tree = renderer.create(<H5>Hello World</H5>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('H6 component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<H6>Hello World</H6>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('H6 component renders unchanged', () => {
  const tree = renderer.create(<H6>Hello World</H6>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('P component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<P>Hello World</P>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('P component renders unchanged', () => {
  const tree = renderer.create(<P>Hello World</P>).toJSON();
  expect(tree).toMatchSnapshot();
});
