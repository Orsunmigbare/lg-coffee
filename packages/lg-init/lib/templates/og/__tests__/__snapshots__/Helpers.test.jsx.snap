// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 component renders unchanged 1`] = `
<h1
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h1>
`;

exports[`H2 component renders unchanged 1`] = `
<h2
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h2>
`;

exports[`H3 component renders unchanged 1`] = `
<h3
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h3>
`;

exports[`H4 component renders unchanged 1`] = `
<h4
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h4>
`;

exports[`H5 component renders unchanged 1`] = `
<h5
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h5>
`;

exports[`H6 component renders unchanged 1`] = `
<h6
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</h6>
`;

exports[`P component renders unchanged 1`] = `
<p
  className=""
  style={
    Object {
      "color": "#212529",
    }
  }
>
  Hello World
</p>
`;
